hudi-io/src/main/java/org/apache/hudi/io/hfile/ChecksumType.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.hudi.io.hfile;
+
+public enum ChecksumType {
+
+  NULL((byte) 0) {
+    @Override
+    public String getName() {
+      return "NULL";
+    }
+  },
+
+  CRC32((byte) 1) {
+    @Override
+    public String getName() {
+      return "CRC32";
+    }
+  },
+
+  CRC32C((byte) 2) {
+    @Override
+    public String getName() {
+      return "CRC32C";
+    }
+  };
+
+  private final byte code;
+
+  public static ChecksumType getDefaultChecksumType() {
+    return ChecksumType.CRC32C;
+  }
+
+  /** returns the name of this checksum type */
+  public abstract String getName();
+
+  private ChecksumType(final byte c) {
+    this.code = c;
+  }
+
+  public byte getCode() {
+    return this.code;
+  }
+
+  /**
+   * Cannot rely on enum ordinals . They change if item is removed or moved. Do our own codes. n
+   * * @return Type associated with passed code.
```

Comment:
nit: improve the docs?
```suggestion
   * Use designated byte value to indicate checksum type.
   * @return Type associated with passed code.
```

Reply:
Done.